src/main/scala/org/constellation/consensus/ResolutionService.scala

```diff
@@ -0,0 +1,109 @@
+package org.constellation.consensus
+
+import java.util.concurrent.TimeUnit
+
+import akka.actor.ActorRef
+import akka.util.Timeout
+import org.constellation.Data
+import org.constellation.primitives.{APIBroadcast, Schema}
+import org.constellation.primitives.Schema.{CheckpointBlock, CheckpointCacheData, SignedObservationEdgeCache}
+
+import scala.annotation.tailrec
+import scala.concurrent.{ExecutionContext, Future}
+
+object ResolutionService {
+  implicit val timeout: Timeout = Timeout(5, TimeUnit.SECONDS)
+
+  case class ResolutionStatus(
+                               cb: CheckpointCacheData,
+                               resolvedParents: Seq[(String, Option[SignedObservationEdgeCache])],
+                               unresolvedParents: Seq[(String, Option[SignedObservationEdgeCache])]
+                             )
+
+  /**
+    * TODO: Need to include signatories ABOVE this checkpoint block later in the case of signature decay. Add to SignedObservationEdgeCache
+    *
+    * @param dao
```

Comment:
IMO you should either describe these params or delete them. As is they are just noise.

Reply:
Planning on it